guides/source/active_storage_overview.md

```diff
@@ -0,0 +1,305 @@
+create a new one, rather than attempt to mutate the existing one (though of
+course you can delete the previous version later if you don't need it).
+
+
+## Setup
+
+To setup an existing application after upgrading to Rails 5.2, run `rails active_storage:install`. If you're creating a new project with Rails 5.2, ActiveStorage will be installed by default. This generates the tables for the
+`Attachment` and `Blob` models.
+
+Inside a Rails application, you can set-up your services through the
+generated `config/storage.yml` file and reference one
+of the aforementioned constant under the +service+ key. For example:
+
+``` yaml
+  local:
+    service: Disk
+    root: <%= Rails.root.join("storage") %>
+```
+NOTE: Should we include the required keys for all the supported services?
+NOTE: Should we mention the mirror service and how to set it up?
+
+In your application's configuration, specify the service to
+use like this:
+
+``` ruby
+config.active_storage.service = :local
+```
+
+Like other configuration options, you can set the service application wide, or per
+environment. For example, you might want development and test to use the Disk
+service instead of a cloud service.
+
+Attach Files to a Model
+--------------------------
+One or more files can be attached to a model.
+
+One attachment:
+
+```ruby
+class User < ApplicationRecord
+  # Associates an attachment and a blob. When the user is destroyed they are
+  # purged by default (models destroyed, and resource files deleted).
+  has_one_attached :avatar
+end
+
+# Attach an avatar to the user.
+user.avatar.attach(io: File.open("/path/to/face.jpg"), filename: "face.jpg", content_type: "image/jpg")
+
+class AvatarsController < ApplicationController
+  def update
+    # params[:avatar] contains a ActionDispatch::Http::UploadedFile object
+    Current.user.avatar.attach(params.require(:avatar))
+    redirect_to Current.user
+  end
+end
+```
+
+Many attachments:
+
+```ruby
+class Message < ApplicationRecord
+  has_many_attached :images
+end
+```
+
+```erb
+<%= form_with model: @message, local: true do |form| %>
+  <%= form.text_field :title, placeholder: "Title" %><br>
+  <%= form.text_area :content %><br><br>
+
+  <%= form.file_field :images, multiple: true %><br>
+  <%= form.submit %>
+<% end %>
+```
+
+```ruby
+class MessagesController < ApplicationController
+  def index
+    # Use the built-in with_attached_images scope to avoid N+1
+    @messages = Message.all.with_attached_images
+  end
+
+  def create
+    message = Message.create! params.require(:message).permit(:title, :content)
+    message.images.attach(params[:message][:images])
+    redirect_to message
+  end
+
+  def show
+    @message = Message.find(params[:id])
+  end
+end
+```
+
+Remove File Attached to Model
+-------------------------------
+
+To remove an attachment from a model, call `purge` on the attachment. Removal
+can be done in the background if your application is setup to use ActiveJob.
```

Comment:
Setup is a noun. The verb form is “set up.”

Use “Active Job” (with a space between the words) to refer to the framework.